src/main/java/frc/robot/subsystems/intake/Intake.kt

```diff
@@ -50,6 +53,14 @@ abstract class Intake : Subsystem {
     }
 
     fun moveIntake(position: Rotation2d, velocity: Rotation2d) {
+        stallDetector.calculate(inputs.armVelocity.radians, position.radians)
+
+        // only reset if we're moving towards the hard stop and not making progress
+        if (stallDetector.i > 0.5 && position.radians - inputs.position.radians > 0.0) {
```

Comment:
im fairly sure accessing stallDetector.i is just getting the I coefficient of the controller, which will always be one.